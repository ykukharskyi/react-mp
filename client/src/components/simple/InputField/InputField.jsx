import React from 'react';

import Input from './InputField.styles';

const InputField = ({ placeholder }) => (
  <Input placeholder={placeholder} />
);

export default InputField;
