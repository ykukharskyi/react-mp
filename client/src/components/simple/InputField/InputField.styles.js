import styled from 'styled-components';

export default styled.input`
  width: 100%;
  height: 30px;
  color: white;
  background-color: black;
  border-width: 0 0 2px 0;
  border-bottom-color: red;
  font-size: 16px;
`;
