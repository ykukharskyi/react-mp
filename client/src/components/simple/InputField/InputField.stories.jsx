import React from 'react';
import { storiesOf } from '@storybook/react';
import InputField from './InputField';

storiesOf('InputField', module)
  .add('default', () => <InputField />)
  .add('with placeholder', () => <InputField placeholder="hi there" />);
