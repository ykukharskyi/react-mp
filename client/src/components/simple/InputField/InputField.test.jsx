import React from 'react';
import { shallow } from 'enzyme';
import InputField from './InputField';

describe('InputField', () => {
  it('default render', () => {
    const wrapper = shallow(<InputField />);
    expect(wrapper).toMatchSnapshot();
  });

  it('with placeholder', () => {
    const wrapper = shallow(<InputField placeholder="hi there" />);
    expect(wrapper).toMatchSnapshot();
  });
});
