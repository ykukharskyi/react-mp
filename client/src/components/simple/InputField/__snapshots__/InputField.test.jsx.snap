// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField default render 1`] = `<styled.input />`;

exports[`InputField with placeholder 1`] = `
<styled.input
  placeholder="hi there"
/>
`;
